src/sshkeys/sshkeys.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma.service';
+import { Prisma, SshKey } from '@prisma/client';
+import { exclude } from 'src/utils/prisma-exclude';
+import { SanitizedSshKey } from './types/sanitized-ssh-key';
+import { CreateSshKeyDto } from './dto/create-sshkey.dto';
+
+@Injectable()
+export class SshKeysService {
+  constructor(
+    private prisma: PrismaService,
+  ) {}
+
+
+  private sanitizeOutput(sshKey: SshKey): SanitizedSshKey {
+    // all fields seem to be okay to return because SSH keys can only
+    // be consulted by either their owner or the administrators
+    // still, we leave the method in case it is needed later
+    return exclude(sshKey, []);
+  }
```

Comment:
Should you sanitize the output? for now it's returning something like this (on the GET method):

```json
[
	{
		"id": "f2e7e61a-3980-416c-bbb4-eb4b4be18a58",
		"name": "a_key_for_something",
		"value": "blabla_value",
		"createdAt": "2023-07-07T21:42:44.991Z",
		"updatedAt": "2023-07-07T21:42:44.991Z",
		"userId": "9b96e7c4-f418-443e-a2a3-d8cde20d376d"
	}
]
```

I think you should remove `userId` and `id`

Reply:
As stated above, the only users which will see this returned element are either the resource owner, meaning the author of this ssh key, and the administrators. The medium-sensitive `userId` is thus already known of both receiving parties.